src/Core/Domain/Order/Exception/OrderConstraintException.php

```diff
@@ -31,4 +31,18 @@
  */
 class OrderConstraintException extends OrderException
 {
+    /**
+     * Used in create order from BO when the customer message is invalid.
+     */
+    const INVALID_CUSTOMER_MESSAGE = 1;
+
+    /**
+     * @param int $code
+     * @param string $message
+     * @param Throwable|null $previous
+     */
+    public function __construct($code = 0, $message = '', Throwable $previous = null)
```

Comment:
Don't change the regular constructor, since this exception is also used elsewhere
When you throw it you can use a message that describes the problem along with the error code as a second argument